### PR TITLE
WD-481 Store and fetch the bakery outside of the Redux store

### PR DESCRIPTION
**Pull request by huwshimi**
## Done

- Move bakery to a separate file.
- Replace selectors and reducers with bakery import.
- Update tests to use the new bakery setup.

## QA

- Pull code
- Run `dotrun clean && dotrun serve`
- Open http://0.0.0.0:8036/
- Check that you can still log in/out and display models.

### CODE REVIEW

File: src/@canonical/macaroon-bakery.d.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+declare module "@canonical/macaroon-bakery";
```
Comment (goulinkh) [inline src/@canonical/macaroon-bakery.d.ts:1]: Sorry I missed your PR on the bakery lib to move to Typescript!

File: src/app/root.js [hunk 2]
```diff
@@ -32,7 +31,7 @@ function rootReducer(state = {}, action) {
         draftState.visitURL = action.payload;
         break;
       case actionsList.logOut:
-        delete draftState.bakery.storage._store.identity;
+        delete bakery.storage._store.identity;
```
Comment (goulinkh) [inline src/app/root.js:34]: I'm not sure why we are doing this at 2 places: here & `actions/logOut`?

**Review by goulinkh — APPROVED** [review]
I suppose we don't need a state for bakery so this makes sense :+1:

File: src/@canonical/macaroon-bakery.d.ts (hunk 1 above)
Comment (huwshimi) [reply src/@canonical/macaroon-bakery.d.ts:1]: No worries, thanks for the review!

File: src/app/root.js (hunk 2 above)
Comment (huwshimi) [reply src/app/root.js:34]: Ah yeah, well spotted. I can't see any reason for that so I'll remove it from here.

**Pull request merged**